lib/Targets/AIETargetCDO.cpp

```diff
@@ -399,6 +399,7 @@ mlir::LogicalResult AIETranslateToCDO(ModuleOp m, raw_ostream &output) {
         packetID = op.getPacketID();
       }
 
+      assert(blockMap.contains(&block));
       int bdNum = blockMap[&block];
       if (foundBd) {
```

Comment:
It looks like `foundBd` is the same as `blockMap.contains(&block)`, and both the assert and the lookup should move inside the if. That might explain the test failure.